fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java

```diff
@@ -89,7 +89,7 @@ public static ColumnRefOperator findOrCreateColumnRefForExpr(Expr expression, Ex
                                                                  ColumnRefFactory columnRefFactory) {
         ColumnRefOperator columnRef;
         ScalarOperator scalarOperator = SqlToScalarOperatorTranslator.translate(expression, expressionMapping);
-        if (expressionMapping.hasExpression(expression) || scalarOperator.isColumnRef()) {
```

Comment:
why change this?

Reply:
`ScalarOperator scalarOperator = SqlToScalarOperatorTranslator.translate(expression, expressionMapping);`
always returns `ColumnRefOperator` if test `expressionMapping.hasExpression(expression)` is positive.
Otherwise, the below cast expression will throw exception.

So the condition is redundant, and I removed it.